File: leet code practice/Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        queue=[]
        arr=[]
        res=[]
        
        if not root:
            return []
           
            
        
        queue.append(root)
        while len(queue)!=0 :
            size=len(queue)
            for i in range(size):
                removed=queue.pop(0)
                arr.append(removed.val)
                if i == size-1 :
                    res.append(arr)
                    arr=[]
                if  removed.left  :
                    queue.append(removed.left)
                if  removed.right :
                    queue.append(removed.right)
        return res
                
        

File: leet code practice/Binary Tree Level Order Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    if(!root)
        return []
    var queue=[]
    var res=[]
    var arr=[]
    queue.push(root)
    while(queue.length!==0){
        var size=queue.length
        for(let i=0;i<size;i++){
            var removed=queue.shift()
            // console.log(removed)
            arr.push(removed.val)
            if(i===size-1){
                res.push(arr)
                arr=[]
            }
            if(removed.left!=null)
                    queue.push(removed.left);
                 if(removed.right!=null)
                    queue.push(removed.right);
                 
        }
    }
    return res
};